File: C++/Level2/위장.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<vector<string>> clothes) {

    int answer = 1;
    unordered_map<string, int> hmap;

    for (vector<string> vect : clothes)
        hmap[vect.at(1)]++;

    vector<int> count(hmap.size());

    for (auto iter : hmap)
        answer += answer * iter.second;

    return answer - 1;
}

File: programmers-coding-test/Python/Level2/위장.py
def solution(clothes):
    answer = 1
    clothes_type = {}
    for c in clothes:
        if c[1] in clothes_type:
            clothes_type[c[1]] += 1
        else:
            clothes_type[c[1]] = 1
    for n in clothes_type.values():
        print(n)
        answer *= (n + 1)
    return answer - 1
